app/models/ext_management_system.rb

```diff
@@ -49,6 +49,7 @@ def self.api_allowed_attributes
   has_many :hardwares,         :through => :vms_and_templates
   has_many :networks,          :through => :hardwares
   has_many :disks,             :through => :hardwares
+  has_many :physical_servers,  :foreign_key => :ems_id, :inverse_of => :ext_management_system, :dependent => :destroy
```

Comment:
This association is already in the [`PhysicalInfraManager`](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/physical_infra_manager.rb#L7).  Was there a reason why you had to add it here?  If so shouldn't you move it and update the [`PhysicalServer`](https://github.com/ManageIQ/manageiq/blob/master/app/models/physical_server.rb#L20-L21) to not need the `:class_name` ?

Reply:
Without this change, the alert condition never gets evaulated when the hourly timer triggers because the following error occurs:

```
[----] E, [2018-07-30T14:26:16.697127 #10124:2b12ed46d108] ERROR -- : [ActiveRecord::HasManyThroughSourceAssociationNotFoundError]: Could not find the source association(s) "physical_server" or :physical_servers in model ExtManagementSystem. Try 'has_many :physical_servers, :through => :ext_management_systems, :source => <name>'. Is it one of custom_button_sets, provider, child_managers, tenant, container_deployments, endpoints, hosts, non_clustered_hosts, clustered_hosts, vms_and_templates, miq_templates, vms, operating_systems, hardwares, networks, disks, storages, ems_events, generated_events, policy_events, blacklisted_events, miq_alert_statuses, ems_folders, ems_clusters, resource_pools, customization_specs, storage_profiles, customization_scripts, iso_datastore, zone, metrics, metric_rollups, vim_performance_states, miq_events, cloud_subnets, compliances, custom_attributes, miq_custom_attributes, taggings, tags, all_relationships, authentications, or vim_performance_operating_ranges?  Method:[block (2 levels) in <class:LogProxy>]
```